addons/l10n_id_pos/models/qris_transaction.py

```diff
@@ -0,0 +1,19 @@
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class QRISTransaction(models.Model):
+    _inherit = "l10n_id.qris.transaction"
+
+    @api.constrains('model')
+    def _constraint_model(self):
+        # add pos.order to list of possible models
+        if self.model != 'pos.order':
+            super()._constraint_model()
```

Comment:
Not incorrect but I think it more proper to add a _get_supported_model() method that returns a list of models.
You can then override it here without touching to the constraint.